src/tools/genpolicy/setup_containerd_docker.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+# Copyright (c) 2023 Microsoft Corporation
+
+# This script aims to adapt the system for the genpolicy tool to be able to use containerd pull function properly.
+# This is needed for managed identity based authentication to private registries using an identity token.
+
+# This script needs 'sudo' access. It will:
+# - install containerd if not installed already. Genpolicy tool uses containerd to pull image when using -d option
+# - ensure cri plugin is NOT disabled in containerd config. This is needed for policy tool to pull image
+# - restart containerd or start it if not running already. 
+# - print containerd socket file location. Use this when running genpolicy tool. Eg genpolicy -d=$SOCKET_FILE_LOCATION -y foo.yaml
+# - fix containerd socket file permissions if needed. This is so genpolicy tool is able to access containerd socket file without 'sudo'
+# - adapt docker config.json if needed. This is done so 'az acr login' command saves identity token to the docker config
+
+set -e -x
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" &> /dev/null
+}
+
+# Utility function for error messages and exit
+error_exit() {
+    echo "$1" 1>&2
+    exit 1
+}
+
+# Function to ensure a command is installed
+ensure_command_installed() {
+    if ! command_exists "$1"; then
+        echo "$1 could not be found, installing..."
+        sudo apt-get install -y "$1"
```

Comment:
Is it practical to add support for Azure Linux too?

Reply:
Added!